src/js/components/Header.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+class Header extends React.Component {
+
+  getIcon(icon, className) {
+    if (icon == null) {
+      return null;
+    }
+
+    return (
+      <div className="media-object-item">
+        <div className={className}>
+          {icon}
+        </div>
+      </div>
+    );
+  }
+
+  getTitle(title, className) {
+    if (title == null) {
+      return null;
+    }
+
+    return (
+      <h1 className={className}>
+        {title}
+      </h1>
+    );
+  }
+
+  getSubTitle(subTitle) {
+    if (subTitle == null) {
+      return null;
+    }
+
+    return (<div>{subTitle}</div>);
+  }
+
+  render() {
+    let props = this.props;
+
+    return (
+      <div className={props.className}>
+        <div className={props.dividerClassName}>
+          <div className={props.mediaWrapperClassName}>
+            <div className="media-object flex-box flex-box-align-vertical-center">
+              {this.getIcon(props.icon, props.iconClassName)}
+              <div className="media-object-item">
+                {this.getTitle(props.title, props.titleClassName)}
+                {this.getSubTitle(props.subTitle)}
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+Header.defaultProps = {
+  className: 'container container-fluid container-pod container-pod-short flush flush-top',
+  mediaWrapperClassName: 'media-object-spacing-wrapper media-object-spacing-narrow media-object-offset',
+  dividerClassName: 'container-pod container-pod-short flush-top container-pod-divider-bottom container-pod-divider-bottom-align-right container-pod-divider-inverse',
+  titleClassName: 'flush inverse',
+  iconClassName: 'icon icon-large icon-image-container icon-app-container',
```

Comment:
Still, deciding whether to use `classNames` or not is bookkeeping really. I feel like a situation where the user needs to manage the internal class structure of a component is a result of a bigger problem - either that the component is too complex and should be broken into smaller pieces, or that the code structure is being dictated by the styles.